internal/services/lighthouse/lighthouse_definition_resource_test.go

```diff
@@ -383,3 +404,50 @@ resource "azurerm_lighthouse_definition" "test" {
 }
 `, data.RandomInteger, secondTenantID, principalID, planName, planPublisher, planProduct, planVersion)
 }
+
+func (LighthouseDefinitionResource) eligibleAuthorization(id string, secondTenantID string, principalID string, data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+data "azurerm_role_definition" "contributor" {
+  role_definition_id = "b24988ac-6180-42a0-ab88-20f7382dd24c" // Contributor role
+}
+
+data "azurerm_role_definition" "reader" {
+  role_definition_id = "acdd72a7-3385-48ef-bd42-f606fba81ae7"
+}
+
+data "azurerm_subscription" "test" {}
+
+resource "azurerm_lighthouse_definition" "test" {
+  lighthouse_definition_id = "%s"
+  name                     = "acctest-LD-%d"
+  managing_tenant_id       = "%s"
+  scope                    = data.azurerm_subscription.test.id
+
+  authorization {
+    principal_id           = "%s"
+    role_definition_id     = data.azurerm_role_definition.reader.role_definition_id
+    principal_display_name = "Reader"
+  }
+
+  eligible_authorization {
```

Comment:
Should we have a test with multiple entries for this new block?

Reply:
As test data limitation, there is no more principal for testing. [As you can see](https://github.com/hashicorp/terraform-provider-azurerm/pull/19499#pullrequestreview-1201643682), we only can test them on local for test data limitation issue.